android-json-form-wizard/src/test/java/com/vijay/jsonwizard/filesource/DiskFileSourceTest.java

```diff
@@ -0,0 +1,86 @@
+package com.vijay.jsonwizard.filesource;
+
+import android.content.Context;
+import android.os.Environment;
+
+import org.jeasy.rules.api.Rules;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStream;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Environment.class})
+public class DiskFileSourceTest {
+
+    @Rule
+    public TemporaryFolder storageDirectory = new TemporaryFolder();
+
+    @Mock
+    private Context context;
+
+    private DiskFileSource diskFileSource = Mockito.spy(DiskFileSource.INSTANCE);
+
+    @Mock
+    private File externalFile;
+
+    @Before
+    public void setUp() {
+        Mockito.doReturn("/tmp/downloads").when(externalFile).toString();
+
+        PowerMockito.mockStatic(Environment.class);
+        Mockito.when(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)).thenReturn(externalFile);
+    }
+
+    @Test
+    public void testGetRulesFromFileConvertsFileToRules() throws Exception {
+
+        String specifiedString = "1";
+        String relevance = "---\n" +
+                "name: step1_last_name\n" +
+                "description: last_name\n" +
+                "priority: 1\n" +
+                "condition: \"step1_first_Name.equalsIgnoreCase('Doe')\"\n" +
+                "actions:\n" +
+                "    - \"calculation = " + specifiedString + "\"";
+
+        InputStream inputStream = new ByteArrayInputStream(relevance.getBytes());
+        Mockito.doReturn(inputStream).when(diskFileSource).getInputStream(Mockito.any(File.class));
+
+        Rules rules = diskFileSource.getRulesFromFile(context, "test_rule");
+        Assert.assertFalse(rules.isEmpty());
```

Comment:
Wouldn't it be better to verify that the content is as expected and not just that it's not empty?

Reply:
The responsibility of the interface is to return the Rules Object when given a rule but not to verify that the Rule object works.  i.e Given a syntactically correct input of type A, you expect an output or type B